core/domain/state_domain_test.py

```diff
@@ -1031,3 +1621,71 @@ def test_validation_with_unknown_langauge_code_raise_error(self):
 
         with self.assertRaisesRegexp(Exception, 'Invalid language_code: ed'):
             recorded_voiceovers.validate(['content'])
+
+    def test_validation_with_invalid_content_id_list(self):
+        recorded_voiceovers_dict = {
+            'voiceovers_mapping': {
+                'content': {
+                    'en': {
+                        'filename': 'xyz.mp3',
+                        'file_size_bytes': 123,
+                        'needs_update': False
+                    }
+                }
+            }
+        }
+
+        recorded_voiceovers = state_domain.RecordedVoiceovers.from_dict(
+            recorded_voiceovers_dict)
+
+        with self.assertRaisesRegexp(
+            Exception,
+            re.escape(
+                'Expected state recorded_voiceovers to match the listed '
+                'content ids [\'invalid_content\']')):
+            recorded_voiceovers.validate(['invalid_content'])
+
+
+class VoiceoverDomainTests(test_utils.GenericTestBase):
+
+    def setUp(self):
+        super(VoiceoverDomainTests, self).setUp()
+        self.voiceover = state_domain.Voiceover('filename.mp3', 10, False)
+        self.voiceover.validate()
+
+    def test_validate_non_str_filename(self):
+        self.voiceover.filename = 0
```

Comment:
Try adding a baseline verification in this kinda tests (ditto below).

Reply:
the baseline verification is done in the setup function

Reply:
Generally don't do that -- setup is setup. Do baseline verification in the test itself.

Reply:
done